.travis.yml

```diff
@@ -1,20 +1,30 @@
 if: (branch = master) OR (type = pull_request) OR (tag IS present)
 language: rust
 rust: 1.39.0
-cache: cargo
-before_cache:
-  - find ./target/debug -type f -maxdepth 1 -delete
-  - rm -fr ./target/debug/{deps,.fingerprint}/{*ra_*,*test*,*tools*,*gen_lsp*}
-  - rm -f  ./target/.rustc_info.json
+cache:
+  directories:
+    - $HOME/.cache/sccache
+    - $HOME/.cargo
```

Comment:
Should probably add this somewhere:
```
before_cache:
  - rm -rf "$TRAVIS_HOME/.cargo/registry/src"
```

It's part of the `cache: cargo` so you don't cache a whole bunch of extra src. Besides, cargo does parallel downloads of source so its probably not much faster to cache it.

https://docs.travis-ci.com/user/languages/rust/#dependency-management